torch/csrc/jit/import_source.cpp

```diff
@@ -194,25 +194,111 @@ struct SourceImporter {
       auto parsed_treeref = p_.parseClassLike();
       if (parsed_treeref->kind() == TK_CLASS_DEF) {
         auto class_def = ClassDef(parsed_treeref);
-        if (class_def.superclass().present()) {
+        bool is_module = class_def.superclass().present();
+        if (is_module && Var(class_def.superclass().get()).name().name() != "Module") {
           throw ErrorReport(class_def.range())
               << "Torchscript does not support class inheritance.";
         }
         const auto qualified_classname = QualifiedName(
             QualifiedName(class_qualifier), class_def.name().name());
+        auto class_type = ClassType::create(
+            c10::QualifiedName(qualified_classname), owner, is_module);
 
-        std::vector<Def> definitions;
+        std::vector<Def> methods;
         std::vector<ResolverPtr> resolvers;
-        for (const auto& method_def : class_def.body()) {
-          definitions.emplace_back(Def(method_def));
-          resolvers.emplace_back(resolver_);
+        std::vector<Assign> attributes;
+
```

Comment:
This function is really long. It has two phases, parsing the statement body and then acting on that parse. Can we split this up?

Reply:
going to follow up on this because there are a bunch of downstream changes